packages/cloudflare/src/cli/templates/worker.ts

```diff
@@ -2,14 +2,16 @@ import { AsyncLocalStorage } from "node:async_hooks";
 import type { IncomingMessage } from "node:http";
 import Stream from "node:stream";
 
-import type { ExportedHandler, Fetcher } from "@cloudflare/workers-types";
 import type { NextConfig } from "next";
 import { NodeNextRequest, NodeNextResponse } from "next/dist/server/base-http/node";
 import { MockedResponse } from "next/dist/server/lib/mock-request";
 import type { NodeRequestHandler } from "next/dist/server/next-server";
 
 import type { CloudflareContext } from "../../api";
 
+// @ts-expect-error: resolved by wrangler build
+import { handler as middlewareHandler } from "./middleware/handler.mjs";
```

Comment:
Instead of a `@ts-expect-error` I think it would be nicer to define a proper `d.ts` file for this (so that we can have proper type safety)

Something like this for example:
![Screenshot 2024-11-04 at 10 31 33](https://github.com/user-attachments/assets/93eea9dc-31e5-401f-a0fd-086aa344643e)

(although I don't really know if the type I'm using there is the correct one, or even if the type we want is actually exported, but you get the idea 🙂)

Reply:
> so that we can have proper type safety

This is **not** type safety but rather akin to a cast, so I'm -1 on integrating this.

`@ts-expect-error` is also what ona uses and I like how this makes things explicit.

Reply:
> This is not type safety but rather akin to a cast, so I'm -1 on integrating this.

Yes I agree that it is a cast, but if we did have the correct type it would introduce type safety to the code using the handler

The `@ts-expect-error` makes it so that `middlewareHandler` is basically an `any` which I think is clearly wrong in terms of type safety

Reply:
> but if we did have the correct type

Which is the exact problem with a cast: by overriding the type there is no way to make sure you have the correct type.

This is "internal" code so I think `@ts-expect-error` is a reasonable thing to do here. If there are breaking changes in the handler the open next build would and it will be caught by the tests _(whether or not there is `.d.ts`)_.